### PR TITLE
#169642557 Build a base server for the application

### PR DESCRIPTION
**Pull request by b0nbon1**
#### What does this PR do?
 It basically creates a base server for the application that is used when the application is running.
#### Description of Task to be completed?
- Separate the entry point of the application and the server.
- create a server port.
#### How should this be manually tested?
- clone the repo
- run `npm install` to install dependencies
- run `npm run watch` to go into debugging mode.
#### Any background context you want to provide?
- N/A
#### What are the relevant pivotal tracker stories?
- [#169642557](https://www.pivotaltracker.com/n/projects/2407417/stories/169642557)

### CODE REVIEW

**Review by RaymondMwaura — DISMISSED** [review]
This looks O.K.

**Review by RaymondMwaura — COMMENTED** [review]
When the server is running and I give the command 'Ctrl + C', the following message is printed in duplicate:

```
info: SIGINT signal received. {"timestamp":"2019-11-11T13:25:04.479Z"}
info: Closing http server. {"timestamp":"2019-11-11T13:25:04.482Z"}
info: Http server closed. {"timestamp":"2019-11-11T13:25:04.483Z"}
info: SIGINT signal received. {"timestamp":"2019-11-11T13:25:04.483Z"}
info: Closing http server. {"timestamp":"2019-11-11T13:25:04.484Z"}
info: Http server closed. {"timestamp":"2019-11-11T13:25:04.484Z"}
```
Also, the error message below is printed out after closing the server using 'Ctrl+C':

```
npm ERR! code ELIFECYCLE
npm ERR! errno 130
npm ERR! express-barefoot-nomad@1.0.0 watch: `npm run winston && nodemon --exec babel-node -r ./node_modules/dotenv/config ./src/index.js`
npm ERR! Exit status 130
npm ERR! 
npm ERR! Failed at the express-barefoot-nomad@1.0.0 watch script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/raymond/.npm/_logs/2019-11-11T13_25_04_511Z-debug.log
```
However, this second error seems to be present in other branches as well and so may not be linked to this PR directly.

**Comment by b0nbon1** [comment]
> When the server is running and I give the command 'Ctrl + C', the following message is printed in duplicate:
> 
> ```
> info: SIGINT signal received. {"timestamp":"2019-11-11T13:25:04.479Z"}
> info: Closing http server. {"timestamp":"2019-11-11T13:25:04.482Z"}
> info: Http server closed. {"timestamp":"2019-11-11T13:25:04.483Z"}
> info: SIGINT signal received. {"timestamp":"2019-11-11T13:25:04.483Z"}
> info: Closing http server. {"timestamp":"2019-11-11T13:25:04.484Z"}
> info: Http server closed. {"timestamp":"2019-11-11T13:25:04.484Z"}
> ```
> 
> Also, the error message below is printed out after closing the server using 'Ctrl+C':
> 
> ```
> npm ERR! code ELIFECYCLE
> npm ERR! errno 130
> npm ERR! express-barefoot-nomad@1.0.0 watch: `npm run winston && nodemon --exec babel-node -r ./node_modules/dotenv/config ./src/index.js`
> npm ERR! Exit status 130
> npm ERR! 
> npm ERR! Failed at the express-barefoot-nomad@1.0.0 watch script.
> npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
> npm ERR! A complete log of this run can be found in:
> npm ERR!     /Users/raymond/.npm/_logs/2019-11-11T13_25_04_511Z-debug.log
> ```
> 
> However, this second error seems to be present in other branches as well and so may not be linked to this PR directly.

I think the error is coming from the Winston logger script  I haven't been able to fix it

File: src/index.js [hunk 1]
```diff
@@ -1,64 +1,25 @@
...
-  message: 'Welcome to Barefoot Nomad'
-}));
-
-// to throw an error to the logger
-app.get('/err', (req, res) => {
-  throw new ErrorHandler('Errror here');
+app.set('port', process.env.PORT || 3000);
+const server = app.listen(app.get('port'), () => {
+  // eslint-disable-next-line no-console
+  console.log(`Express running → PORT ${server.address().port}`);
```
Comment (timolinn) [inline src/index.js]: Use the `logger.info`  here. You can remove the eslint disable comment after that.

File: src/app.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
+/* eslint-disable no-unused-vars */
+/* eslint-disable no-console */
```
Comment (timolinn) [inline src/app.js]: Remove unnecessary eslint comments.

File: src/index.js [hunk 3]
```diff
@@ -1,55 +1,18 @@
 /* eslint-disable no-unused-vars */
 /* eslint-disable no-console */
```
Comment (timolinn) [inline src/index.js]: Remove unnecessary eslint comments

**Review by timolinn — CHANGES_REQUESTED** [review]

File: src/index.js [hunk 4]
```diff
@@ -1,60 +1,16 @@
...
-});
-
-// Production and testing error handler middleware
-app.use((err, req, res, next) => {
-  logger.error(`${err.status || 500} - ${err.message} - ${req.originalUrl} - ${req.method} - ${
-    req.ip
-  } - Stack: ${err.stack}`);
-
-  return Responses.handleError(err.statusCode, err.message, res);
+app.set('port', process.env.PORT || 3000);
```
Comment (Mastel22) [inline src/index.js]: we are not making direct calls anymore. You can change to config.PORT

**Review by Mastel22 — DISMISSED** [review]
Good now.

File: src/index.js (hunk 3 above)
Comment (b0nbon1) [reply src/index.js]: fixed now, thanks

File: src/app.js [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+app.use(urlencoded({ extended: false }));
+app.use(json());
+
+// used in testing heroku
+app.get('/', (req, res) => Responses.handleSuccess(200, 'Welcome to Barefoot Nomad', res));
+
+// to throw an error to the logger
+// eslint-disable-next-line no-unused-vars
+app.get('/err', (req, res) => {
```
Comment (timolinn) [inline src/app.js]: What is this middleware for?

Comment (b0nbon1) [reply src/app.js]: I was testing if `throw new error` works

**Review by RaymondMwaura — DISMISSED** [review]
This looks okay to me.

File: src/app.js (hunk 5 above)
Comment (b0nbon1) [reply src/app.js]: @timolinn  should I remove it

Comment (timolinn) [reply src/app.js]: Yes please, since there's no other use for it. Its #DeadCode.

Comment (b0nbon1) [reply src/app.js]: okaay thanks.

**Review by Mastel22 — APPROVED** [review]
LGTM

**Review by timolinn — APPROVED** [review]

**Review by RaymondMwaura — APPROVED** [review]
Good job.

**Pull request merged**